### PR TITLE
Add support for civetweb

### PR DESCRIPTION
**Pull request by leseb**
We prefer the use of civetweb against Apache. So civetweb is enabled by
default.
The container can be run like this:

docker run -d --net=host -e RGW_NAME=myrgw -e RGW_WEB_ENABLE=0 -e
RGW_CIVETWEB_PORT=80 -v /etc/ceph:/etc/ceph ceph/radosgw

The port can be changed with `RGW_CIVETWEB_PORT`.
To enable Apache use: RGW_WEB_ENABLE=1

Closes: #26

Signed-off-by: Sébastien Han <sebastien.han@enovance.com>

### CODE REVIEW

**Comment by leseb** [comment]
Let me rebase this first...

**Comment by leseb** [comment]
Looks like we have an issue on firefly with config, work great with giant though.
@yehudasa any idea?
@Ulexus perhaps we need to merge this first: #32

**Comment by Ulexus** [comment]
I think I'd go in order: #31 then #32 then #33

**Comment by leseb** [comment]
@Ulexus how does this look now?

File: radosgw/README.md [hunk 1]
```diff
@@ -16,3 +17,13 @@ It will look for `/etc/ceph/ceph.client.admin.keyring` with which to authenticat
...
 Commonly, you will want to bind-mount your host's `/etc/ceph` into the container.  For example:
 `docker run -d --net=host -e RGW_NAME=myrgw -v /etc/ceph:/etc/ceph ceph/radosgw`
+
+To enable an external CGI interface instead of civetweb set:
+
+* `RGW_REMOTE_CGI=1`
+* `RGW_REMOTE_CGI_HOST=192.168.0.1`
+* `RGW_REMOTE_CGI_PORT=9000`
+
+And run the container like this `docker run -d --net=host -e RGW_NAME=myrgw -e RGW_REMOTE_CGI=1 -e RGW_REMOTE_CGI_HOST=192.168.0.1 -e RGW_REMOTE_CGI_PORT=9000 -v /etc/ceph:/etc/ceph ceph/radosgw
```
Comment (Ulexus) [inline radosgw/README.md]: Do we actually need `--net=host` for radosgw?

Comment (leseb) [inline radosgw/README.md]: I don't think so since the rgw can reach the mon on the net host.

**Comment by Ulexus** [comment]
It looks good to me.  I can't try it right now, thanks to my ceph cluster deciding to go mad and corrupt half my RBDs, but I have tested the build.

File: radosgw/README.md [hunk 2]
```diff
@@ -15,4 +16,14 @@ For example:
...
-`docker run -d --net=host -e RGW_NAME=myrgw -v /etc/ceph:/etc/ceph ceph/radosgw`
+`docker run -d -e RGW_NAME=myrgw -v /etc/ceph:/etc/ceph ceph/radosgw`
+
+To enable an external CGI interface instead of civetweb set:
+
+* `RGW_REMOTE_CGI=1`
+* `RGW_REMOTE_CGI_HOST=192.168.0.1`
+* `RGW_REMOTE_CGI_PORT=9000`
+
+And run the container like this `docker run -d -e RGW_NAME=myrgw -e RGW_REMOTE_CGI=1 -e RGW_REMOTE_CGI_HOST=192.168.0.1 -e RGW_REMOTE_CGI_PORT=9000 -v /etc/ceph:/etc/ceph ceph/radosgw
```
Comment (Ulexus) [inline radosgw/README.md]: It _should_ have `-p 80` and `-p 9000` though, I imagine.

Comment (leseb) [inline radosgw/README.md]: Well it depends which use case you're in.
This implementation doesn't support multi-backends, should we?

Comment (Ulexus) [inline radosgw/README.md]: Sorry, yes, that should have read `-p 80` _OR_ `-p 9000`.

I have no idea if there is demand for multiple backends or not.  I think I'd merge this and add the feature later, if needed or requested.

**Comment by Ulexus** [comment]
That's also a much faster startup than the apache version.

**Comment by leseb** [comment]
Héhé I can imagine :)

**Pull request merged**